### PR TITLE
Return error instead of crashing when receiving grpc_proxy request on uninitialized daprd

### PR DESCRIPTION
**Pull request by wcs1only**
# Description

If a grpc proxy request comes in before daprd has completed initialization (for instance if daprd is blocked waiting on the app to start listening), daprd will crash. This change returns an error instead.

## Issue reference

Please reference the issue this PR will close: #3982
## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/messaging/grpc_proxy.go [hunk 1]
```diff
@@ -98,6 +98,10 @@ func (p *proxy) intercept(ctx context.Context, fullName string) (context.Context
 		return outCtx, conn, cErr
 	}
 
+	if p.remoteAppFn == nil {
```
Comment (yaron2) [inline pkg/messaging/grpc_proxy.go:101]: Please move the if block before line 83: https://github.com/dapr/dapr/blob/a782b85439c7d3ed87a76b2eb086e2f9eedb89b4/pkg/messaging/grpc_proxy.go#L83.

**Review by yaron2 — CHANGES_REQUESTED** [review]

**Comment by yaron2** [comment]
Left one comment, but would also be good to add a unit test for this when `remoteAppFn` is nil.

File: pkg/messaging/grpc_proxy.go (hunk 1 above)
Comment (wcs1only) [reply pkg/messaging/grpc_proxy.go:101]: Oh, bad rebase from release-1.5 to master. Good catch.

**Comment by wcs1only** [comment]
> Left one comment, but would also be good to add a unit test for this when `remoteAppFn` is nil.

Indeed. Will add.

**Review by yaron2 — APPROVED** [review]
lgtm
